reader.go

```diff
@@ -194,9 +228,13 @@ func (r *Reader) openDbf() (err error) {
 		return
 	}
 
-	r.dbf, err = os.Open(r.filename + ".dbf")
-	if err != nil {
-		return
+	if r.memoryData != nil {
```

Comment:
IMHO this is a bit of a code smell: Testing whether one of the members is holding data and then branching the logic from there.